tracks/rust/exercises/bob/mentoring.md

```diff
@@ -0,0 +1,140 @@
+### Concepts
+
+- [bool](https://doc.rust-lang.org/std/primitive.bool.html)
+- [char](https://doc.rust-lang.org/std/char/index.html)
+- [match statement](https://doc.rust-lang.org/1.2.0/book/match.html)
+- [variables](https://doc.rust-lang.org/book/ch03-01-variables-and-mutability.html)
+- [closures](https://doc.rust-lang.org/book/ch13-01-closures.html)
+- [&str](https://doc.rust-lang.org/std/primitive.str.html)
+- [Iterators](https://doc.rust-lang.org/std/iter/trait.Iterator.html)
+- [Iterator Methods](https://doc.rust-lang.org/std/iter/trait.Iterator.html#provided-methods)
+- [constant](https://doc.rust-lang.org/std/keyword.const.html)
+
+### Reasonable Solutions
+
+A reasonable solution should:
+
+- Use constants
+- Use variables
+- Use `match` statements
+- Check if the input string is empty and return the "Sure" constant
+before doing any additional processing
+- Use simple methods from the standard library
+- Not use the regex crate
+- Not use separate functions
+
+#### Examples
+
+```rust
+const SURE: &str = "Sure.";
+const WHOA: &str = "Whoa, chill out!";
+const CALM: &str = "Calm down, I know what I'm doing!";
+const FINE: &str = "Fine. Be that way!";
+const WHATEVER: &str = "Whatever.";
+
+#[allow(clippy::match_bool)]
+pub fn reply(message: &str) -> &str {
+    let trimmed: &str = message.trim();
+
+    match trimmed.is_empty() {
+        true => FINE,
+        false => {
+            let is_question: bool = trimmed.ends_with("?");
```

Comment:
This should maybe noted. People are often using `chars().last()`, which works as well of course, but `ends_with` is much clearer.

Reply:
Thanks! We actually have it in the "Common Suggestions" section below, but I've clarified it a bit more :)